packages/ngx-deploy-npm/src/executors/deploy/engine/engine.spec.ts

```diff
@@ -128,4 +129,120 @@ describe('engine', () => {
       expect(setPackage.setPackageVersion).not.toHaveBeenCalled();
     });
   });
+
+  describe('Package Version Check Feature', () => {
+    const mockPackageJson = {
+      name: '@test/package',
+      version: '1.0.0',
+    };
+
+    beforeEach(() => {
+      jest
+        .spyOn(fs, 'readFileSync')
+        .mockReturnValue(JSON.stringify(mockPackageJson));
+    });
```

Comment:
We use [sifers](https://medium.com/@kolodny/testing-with-sifers-c9d6bb5b362) as a pattern to write unit tests. Creating this `beforeEach` is against that pattern.

The mocks should be on the primary `setup` function or in a custom function `setup` function.

Reply:
should be good now